src/language/records.md

```diff
@@ -0,0 +1,120 @@
+---
+title: Records
+description: Summary of the record data structure in Dart.
+toc: false
+---
+
+Records are an anonymous, immutable, aggregate type. Like other [collection types][], 
+they let you bundle multiple objects into a single object. Unlike other collection 
+types, records are fixed-sized, heterogeneous, and typed.
+
+Records are real values; you can store them in variables, pass them to and from
+functions, and store them in lists.
+
+## Record syntax
+
+Records are comma-delimited lists of named or positional fields,
+enclosed in parenthesis:
+
+```dart
+var record = ('first', a: 2, b: true, 'last');    // Record expression
+```
+
+Named fields inside record type annotations go inside a brace-delimited section
+of type and name pairs:
+
+```dart
+(Sring, {int a, bool b}, String?) record;    // Record type annotation
+```
+Naming fields in record type annotations is optional for documentation purposes.
```

Comment:
The names of positional fields are optional and only for documentation, but the names of named fields (I admit that's confusing) are not optional. So these two record types are equivalent as far as the type checker is concerned:

```dart
(int x, int y, int z) point = (1, 2, 3);
(int r, int g, int b) color = (1, 2, 3);

// OK:
point = color;
```

But these are not:

```dart
({int x, int y, int z}) point = (x: 1, y: 2, z: 3);
({int r, int g, int b}) color = (r: 1, g: 2, b: 3);

// Error, different types:
point = color;
```

(This is the same for function parameters, for what it's worth.)

Reply:
Oh ok, thanks for explaining. I might just use this example to illustrate that point.

Reply:
Just realized, the exact opposite of what I wrote in the next line 😅 good thing you caught that!

Reply:
@munificent Actually, I'm still not sure I understand. In the first example, if I were to say `print(point.x)` or `print(color.b)`, it would work and print `1` and `3`, right?

And what if you were to write:
```
(int x, int y, int z) point = (a: 1, b: 2, c: 3);
```
That wouldn't work, right?

Reply:
Nevermind, you can ignore all of this. I was somehow reading "the names _of named_ fields" as just "the names of fields" (which was very confusing!), but now I get it.